bitmovin-analytics-datasource/src/utils/dataUtils.test.ts

```diff
@@ -0,0 +1,269 @@
+import {
+  padAndSortTimeSeries,
+  transformGroupedTimeSeriesData,
+  transformSimpleTimeSeries,
+  transformTableData,
+} from './dataUtils';
+import { FieldType } from '@grafana/data';
+
+describe('padAndSortTimeSeries', () => {
+  it('should return sorted and padded data for simple time series data', () => {
+    //arrange
+    const data = [
+      [1712919600000, 2], //Friday, 12 April 2024 11:00:00
+      [1712919720000, 5], //Friday, 12 April 2024 11:02:00
+    ];
+
+    const startTimestamp = 1712919540000; //Friday, 12 April 2024 10:59:00
+    const endTimestamp = 1712919780000; //Friday, 12 April 2024 11:03:00
+
+    //act
+    const result = padAndSortTimeSeries(data, startTimestamp, endTimestamp, 'MINUTE');
+
+    //assert
+    expect(result).toEqual([
+      [1712919540000, 0],
+      [1712919600000, 2],
+      [1712919660000, 0],
+      [1712919720000, 5],
+      [1712919780000, 0],
+    ]);
+  });
+
+  it('should return sorted and padded data for grouped time series data', () => {
+    //arrange
+    const data = [
+      [1712919600000, 'BROWSER', 'DEVICE_TYPE', 2], //Friday, 12 April 2024 11:00:00
+      [1712919720000, 'BROWSER', 'DEVICE_TYPE', 5], //Friday, 12 April 2024 11:02:00
```

Comment:
You are right, it will only fill the data with the first group by entries. That is desired behaviour, this method is only about padding the data and not about grouping it.  The grouping needs to happen before calling the padding function. So when we call the padding function we already have a timeseries that consists of only the same group by values. 
I can add this to the java doc for the method.